vnext/Mso/src/dispatchQueue/looperScheduler.cpp

```diff
@@ -44,35 +44,37 @@ LooperScheduler::~LooperScheduler() noexcept {
   for (;;) {
     if (auto self = weakSelf.GetStrongPtr()) {
       if (auto queue = self->m_queue.GetStrongPtr()) {
-        DispatchQueue q{Mso::CntPtr(queue)};
+        auto &taskStarting = self->m_settings.TaskStarting;
+        auto &taskCompleted = self->m_settings.TaskCompleted;
         DispatchTask task;
         while (queue->TryDequeTask(task)) {
-          if (auto &func = self->m_settings.TaskStarting) {
-            func(q);
+          if (taskStarting) {
+            taskStarting(DispatchQueue{Mso::CntPtr(queue)});
           }
-
           queue->InvokeTask(std::move(task), std::nullopt);
-
-          if (auto &func = self->m_settings.TaskCompleted) {
-            func(q);
+          if (taskCompleted) {
+            taskCompleted(DispatchQueue{Mso::CntPtr(queue)});
           }
         }
+      }
 
-        if (self->m_isShutdown) {
-          break;
-        }
+      if (self->m_isShutdown) {
+        break;
+      }
 
-        if (auto &func = self->m_settings.IdleWaitStarting) {
-          func(q);
+      if (auto &func = self->m_settings.IdleWaitStarting) {
+        if (auto queue = self->m_queue.GetStrongPtr()) {
+          func(DispatchQueue{Mso::CntPtr(std::move(queue))});
```

Comment:
>Mso::CntPtr [](http://example.com/codeflow?start=29&length=11)

There is no need to create Mso::CntPtr in this context. DispatchQueue needs Mso::CntPtr&& and the std::move provides it. I.e. it should be:
```c++
func(DispatchQueue{std::move(queue)});
```

Reply:
Done!